### PR TITLE
refactor(credentials): Don't rely on explicit registration of CredentialProviders

### LINKED ISSUE #569: reset-db should clean also data for api/v1/connectors/twitter/credentials endpoint
When there are no settings/twitter secrets,

entpoint https://syndesis-qe.b6ff.rh-idev.openshiftapps.com/api/v1/connectors/twitter/credentials
returns empty json `{}`

once setup, this endpoint returns
```json
{"description":"twitter","icon":"twitter","label":"twitter","type":"OAUTH1"}
```

Then I call Reset db, and expect endpoint to return empty json again,
but it still returns 
```json
{"description":"twitter","icon":"twitter","label":"twitter","type":"OAUTH1"}
```

We can't proceed with e2e tests unless we're able to flush state of app!

CC @dsimansk @mmelko @lvydra

### PR DESCRIPTION
**Pull request by zregvart**
Based on #572 so take a look at that first (split to help the review)

This removes explicit registration of CredentialProviders and uses the
database as the source for configuring the CredentialProviders. Also
removes the magical auto-configuration via properties -
CredentialProviders will be configured only through the database so
there is no need for that.

Fixes #569

### CODE REVIEW

**Comment by jimmidyson** [comment]
retest this please

File: credential/src/main/java/io/syndesis/credential/CredentialProviderRegistry.java [hunk 1]
```diff
@@ -16,20 +16,54 @@
...
+
+        credentialProviderFactories = SpringFactoriesLoader
+            .loadFactories(CredentialProviderFactory.class, ClassUtils.getDefaultClassLoader()).stream()
+            .collect(Collectors.toMap(CredentialProviderFactory::id, Function.identity()));
     }
 
     @Override
     public CredentialProvider providerWithId(final String providerId) {
-        final CredentialProvider providerWithId = providers.get(providerId);
+        final Connector connector = dataManager.fetch(Connector.class, providerId);
```
Comment (jimmidyson) [inline credential/src/main/java/io/syndesis/credential/CredentialProviderRegistry.java]: This seems odd to fetch the `Connector` using `providerId`. Is this method really better called `providerForConnector` and `providerId` -> `connectorId`?

Comment (jimmidyson) [reply credential/src/main/java/io/syndesis/credential/CredentialProviderRegistry.java]: Ah the `providerId === credentialsId` always?

Comment (zregvart) [reply credential/src/main/java/io/syndesis/credential/CredentialProviderRegistry.java]: Yeah I've stumbled on that myself, provider comes from Spring Social and we use Connector to name the same thing in this context. I could do a PR to switch all usages of `provider` to `connector`, would that be OK?

Comment (jimmidyson) [reply credential/src/main/java/io/syndesis/credential/CredentialProviderRegistry.java]: Perhaps just raise an issue for it now, it's not urgent, just slightly confusing naming.

Comment (jimmidyson) [reply credential/src/main/java/io/syndesis/credential/CredentialProviderRegistry.java]: Will merge this as-is anyway - thanks!

**Pull request merged**